### PR TITLE
Mongo DB gets updated with the fee after a BTC-like transaction is co…

### PR DESCRIPTION
**Pull request by aspasia**
…nfirmed

@jaromil can you please have a look?

### CODE REVIEW

File: src/social_wallet_api/handler.clj [hunk 1]
```diff
@@ -293,15 +300,32 @@ Creates a transaction.
...
-                                                       (assoc :transaction-id transaction-id)
-                                                       (assoc :currency (:blockchain query))))
+                           (do
+                             ;; Update fee to db when confirmed
+                             ;; The logged-future will return an exception which otherwise would be swallowed till deref
+                             (log/logged-future
+                              (while (> (:number-confirmations blockchain)
+                                        (number-confirmations blockchain transaction-id))
+                                 (log/debug "Not enough confirmations for transaction with id " transaction-id)
+                                 (Thread/sleep 20000))
```
Comment (jaromil) [inline src/social_wallet_api/handler.clj]: let's make also this number a configuration directive like `frequency-confirmations`

File: src/social_wallet_api/handler.clj [hunk 2]
```diff
@@ -293,15 +300,32 @@ Creates a transaction.
...
-                                                   (:to-id query)
-                                                   (-> query 
-                                                       (dissoc :comment :comment-to)
-                                                       (assoc :transaction-id transaction-id)
-                                                       (assoc :currency (:blockchain query))))
+                           (do
+                             ;; Update fee to db when confirmed
+                             ;; The logged-future will return an exception which otherwise would be swallowed till deref
+                             (log/logged-future
+                              (while (> (:number-confirmations blockchain)
```
Comment (jaromil) [inline src/social_wallet_api/handler.clj]: what happens to blockchains that have no need to check confirmations? (like `mongo` or `in-memory`)?
shouldn't there be a flag checked?

**Comment by jaromil** [comment]
Its using freecoin-lib `0.7.0-SNAPSHOT` among dependencies, but is not yet uploaded to clojars.
The rest seems sane, but haven't tested yet.

File: src/social_wallet_api/handler.clj (hunk 1 above)
Comment (aspasia) [reply src/social_wallet_api/handler.clj]: will do

File: src/social_wallet_api/handler.clj (hunk 2 above)
Comment (aspasia) [reply src/social_wallet_api/handler.clj]: This is actually checked above. These are actions only for blockchains other than mongo. I will make the code clearer. I have tested and works for both

**Comment by aspasia** [comment]
The version is a snapshot cause the lib is not yet released. I will make a pr for that first and then release 0.7.0, and then make a new PR. I close this one

**Pull request closed without merging**